internal/commands/pull/inputs.go

```diff
@@ -109,3 +118,46 @@ func (i *inputs) resolveRemoteApp(ui terminal.UI, clients cli.Clients) (realm.Ap
 
 	return app, nil
 }
+
+func (i *inputs) resolveTemplate(ui terminal.UI, realmClient realm.Client, groupID, appID string) (*zip.Reader, []string,  error) {
+	if i.TemplateID == "" {
+		return nil, nil, nil
+	}
+
+	var paths []string
+	if !ui.AutoConfirm() {
+		selectablePaths := []string{
+			frontendPath,
+			backendPath,
+		}
+		if err := ui.AskOne(
+			&paths,
+			&survey.MultiSelect{
+				Message: "Where would you like to export the template?",
+				Options: selectablePaths,
+			}); err != nil {
+			return nil, nil, err
+		}
+	} else {
+		paths = append(paths, frontendPath)
+	}
+
+	compatibleTemplates, err := realmClient.CompatibleTemplates(groupID, appID)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	for _, template := range compatibleTemplates {
+		// This is the ID that they selected
+		if template.ID == i.TemplateID {
```

Comment:
[question] do we ever give the user the ability to _select_ the template id?  in `app create`, we more or less always prompt them for a selection (unless `-y` is set).  especially with things like IDs, i think its helpful to have a way where the user does not need to know the specific value, but can otherwise pick it

Reply:
@sumedham  Does this look fine from a design perspective? 
We're going with a new flow of work as Andrew detailed above

Reply:
this sgtm